### PR TITLE
Add a note about enabling time series index mode via a component template

### LINKED ISSUE #109149: [index.mode=time_series] requires a non-empty [index.routing_path] when creating component template
### Elasticsearch Version

8.13.3

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

Ubuntu

### Problem Description

When trying to create a component template which should basically only set the `index.mode: "time_series"`, this results in

```
[index.mode=time_series] requires a non-empty [index.routing_path]
```

I know, that `index.mode: "time_series"` needs further settings (that might need to come from other component templates).

You get the same error, when you only try to create a component template, only setting `index.routing_path` (without specifying `index.mode: "time_series"` **within** this component template.)

This sounds like an error to me. It should be possible to set those settings or mapping separately in different component templates.

You could have several different component templates like this:

```
PUT _component_template/enable_timeseries_mode
{
  "template": {
    "settings": {
      "index": {
        "mode": "time_series"
      }
    }
  }
}

# 
PUT _component_template/prometheus_dimensions
{
  "template": {
    "settings": {
      "index": {
        "routing_path": [
          "prometheus.labels.*"
        ]
      }
    },
    "mappings": {
      "dynamic_templates": [
        {
          "prometheus_dimensions": {
            "path_match": [
              "prometheus.labels.*"
            ],
            "mapping": {
              "type": "keyword",
              "time_series_dimension": true
            }
          }
        }
      ]
    }
  }
}
```

and use both component templates in the final index-template

```
{
...
  "composed_of": [
    "enable_timeseries_mod",
    "prometheus_dimensions"
  ]
}
```

It is - somehow - related to
- https://github.com/elastic/elasticsearch/issues/96330 and
- https://github.com/elastic/elasticsearch/issues/91592

### Steps to Reproduce

Try to create a new component template

```
PUT _component_template/enable_timeseries_mode
{
  "template": {
    "settings": {
      "index": {
        "mode": "time_series"
      }
    }
  }
}
```

results in

```
{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "[index.mode=time_series] requires a non-empty [index.routing_path]"
      }
    ],
    "type": "illegal_argument_exception",
    "reason": "[index.mode=time_series] requires a non-empty [index.routing_path]"
  },
  "status": 400
}
```

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by martijnvg**
Closes #109149

### CODE REVIEW

File: docs/reference/data-streams/set-up-tsds.asciidoc [hunk 1]
```diff
@@ -279,6 +279,21 @@ POST metrics-weather_sensors-dev/_rollover
 // TEST[setup:tsds]
 // TEST[teardown:tsds_cleanup]
 
+[discrete]
+[[set-up-component-templates]]
+==== A note about component templates and index.mode setting
+
+Configuring a TSDS via an index templates that uses component templates is a bit more complicated.
```
Comment (lkts) [inline docs/reference/data-streams/set-up-tsds.asciidoc:286]:
```suggestion
Configuring a TSDS via an index template that uses component templates is a bit more complicated.
```

File: docs/reference/data-streams/set-up-tsds.asciidoc [hunk 2]
```diff
@@ -279,6 +279,21 @@ POST metrics-weather_sensors-dev/_rollover
...
+With component templates typically mappings and settings get scattered across multiple component templates.
```
Comment (lkts) [inline docs/reference/data-streams/set-up-tsds.asciidoc:287]:
```suggestion
Typically with component templates mappings and settings get scattered across multiple component templates.
```

File: docs/reference/data-streams/set-up-tsds.asciidoc [hunk 3]
```diff
@@ -279,6 +279,21 @@ POST metrics-weather_sensors-dev/_rollover
...
+When configuring the `index.mode` setting in a component template, then the index.routing_path setting needs to
+be defined as well in the same component template. Additionally the fields mentioned in the `index.routing_path`
```
Comment (lkts) [inline docs/reference/data-streams/set-up-tsds.asciidoc:289]:
```suggestion
When configuring the `index.mode` setting in a component template, the index.routing_path setting needs to
be defined in the same component template. Additionally the fields mentioned in the `index.routing_path`
```

File: docs/reference/data-streams/set-up-tsds.asciidoc [hunk 4]
```diff
@@ -279,6 +279,21 @@ POST metrics-weather_sensors-dev/_rollover
...
+also need to be defined in the same component template with the `time_series_dimension` attribute enabled.
+
+The reasons for this is that each component template needs to be valid on its own and the time series index mode
+requires the `index.routing_path` setting. When configuring the  `index.mode` setting in an index template, then
+the `index.routing_path` setting is configured automatically. The `index.routing_path` setting is then derived  from
```
Comment (lkts) [inline docs/reference/data-streams/set-up-tsds.asciidoc:294]:
```suggestion
requires the `index.routing_path` setting. When configuring the  `index.mode` setting in an index template, the `index.routing_path` setting is configured automatically. It is derived  from
```

File: docs/reference/data-streams/set-up-tsds.asciidoc [hunk 5]
```diff
@@ -279,6 +279,21 @@ POST metrics-weather_sensors-dev/_rollover
...
+the field mappings with `time_series_dimension` attribute enabled.
```
Comment (lkts) [inline docs/reference/data-streams/set-up-tsds.asciidoc:295]: So the reason it fails is that we don't see mappings when component template is created? Should we add a sentence to explain that here?

**Review by lkts — APPROVED** [review]

File: docs/reference/data-streams/set-up-tsds.asciidoc (hunk 5 above)
Comment (martijnvg) [reply docs/reference/data-streams/set-up-tsds.asciidoc:294]: No, the logic that automatically defines the routing_path setting is only activated with index templates. The main reason for that is then there is complete view of all the mappings. We could also do this component templates, but this doesn't happen today.

**Pull request merged**